docs/concepts/models.md

```diff
@@ -183,6 +183,18 @@ print(m.model_dump())
 
 For self-referencing models, see [postponed annotations](postponed_annotations.md#self-referencing-or-recursive-models).
 
+!!! note
+    **When defining your models, watch out for naming collisions between your field name and its type, a previously defined model, or an imported library.** 
+
+    For example, the following would yield a validation error:
+    ```
+    class Boo(BaseModel):
+        int: Optional[int] = None
+
+    m = Boo(int=123)
+    ```
+    An error occurs since the field  ```int``` is set to a default value of ```None``` and has the exact same name as its type, so both are interpreted to be ```None```. As such, ```123``` is not considered to be a valid value for the field ```int```.  
```

Comment:
Looks great! I think you can just use single ticks for the code here, like "`" instead of "```".

Also, maybe we go with a slightly more simple example like `int: int`, just for clarity?

Reply:
Updated to use single ticks!

When I was trying out examples to potentially add to the documentation, the following example worked fine in Pydantic:
```
from pydantic import BaseModel

class M1(BaseModel):
   int: int

m = M1(int=123) 
print(repr(m)) # prints M1(int=123) without any error
```
So, it seems that the error occurs when the default value of the field ends up being None, which I think is most feasibly achieved using `Optional[]`?